### PR TITLE
[ADR] Initial technical implementation of USSF portal

### LINKED ISSUE #26: Write ADR on initial technical implementation of new USSF portal
Write an ADR describing our technical approach to starting development of the new USSF portal against the existing MVP.

### PR DESCRIPTION
**Pull request by suzubara**
## Description

This PR adds our first ADR, as well as the standard Truss [ADR template](https://adr.github.io/madr/#the-template). The ADR describes our chosen approach for beginning technical work on the new USSF portal.

Closes #26

### CODE REVIEW

File: docs/adr/0001-initial-portal-implementation.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+- **Prototype:** A sandbox codebase (https://github.com/USSF-ORBIT/ussf-portal-prototype) primarily used to experiment, test, and validate ideas and patterns before committing them to the actual application. This application will never contain or interact with real data, it will never utilize real user authentication, it will never persist data, and it will never evolve into a production codebase.
+
+## Decision Drivers
+
+* We want to be able to test changes with users in a real-life environment with real data
+* We want to get feedback through analytics data and other means quickly and often
+* We don't want to put users through a steep learning curve as they adapt to the new portal
+* We don’t want the use of the portal MVP as a starting point for iterative changes to constrain our own creativity
+* We want to have a significant moment of "launching" the new portal once our new ideas have been validated
```
Comment (esacteksab) [inline docs/adr/0001-initial-portal-implementation.md:25]: You options below address the above list, but none address this one explicitly, though I think `Build the new portal in a complete separate application from the MVP` is the only one that allows for this.

File: docs/adr/0001-initial-portal-implementation.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+* We spend some overhead time and effort re-implementing a website we know will be undergoing significant changes over the course of this project.
+
+## Pros and Cons of the Options
+
+### Implement small, incremental changes to the existing MVP codebase without significant architectural changes
+
+* Good, because small, incremental changes indicate low risk and a fast feedback loop.
+* Good, because we don't have to spend any time re-implementing the existing MVP code.
+* Bad, because we anticipate wanting to make significant changes to the tech stack of the portal to support more complex functionality.
+* Bad, because users may not notice or pay attention to many small changes made over a long period of time.
```
Comment (esacteksab) [inline docs/adr/0001-initial-portal-implementation.md:57]: You call this out as `Bad` here, but wouldn't the same be true for the above (the chosen path)? If you iterate on the current design, are users likely to pay attention to many small changes we make over the duration of the engagement?

**Comment by suzubara** [comment]
@esacteksab I just added some points that I hope will clarify your questions! With the chosen option, we would be implementing something like feature flags and/or A/B tests, which will still allow us to have a "launch" moment by way of turning the new site "on" for everyone instead of a select few. We'll also be able to implement dramatic changes within the new portal code during the experimental phase.

File: docs/adr/0001-initial-portal-implementation.md (hunk 2 above)
Comment (abbyoung) [reply docs/adr/0001-initial-portal-implementation.md:60]: My understanding was that we won't direct all traffic to the new portal/features until there's a significant amount completed and we're ready for a launch. The iterating/testing would be with beta users, and most users would still use the existing portal as-is. Definitely need some clarification cause I could be misremembering! @suzubara

Comment (abbyoung) [reply docs/adr/0001-initial-portal-implementation.md:60]: Never mind I just saw Suz's response to this below 😂 🙄 *leaves*

**Comment by esacteksab** [comment]
> @esacteksab I just added some points that I hope will clarify your questions! With the chosen option, we would be implementing something like feature flags and/or A/B tests, which will still allow us to have a "launch" moment by way of turning the new site "on" for everyone instead of a select few. We'll also be able to implement dramatic changes within the new portal code during the experimental phase.

Ah! I missed the feature flag component and thought we were just doing this out in the open! Sounds good to me! Thanks!

**Review by esacteksab — APPROVED** [review]

**Review by abbyoung — APPROVED** [review]
lgtm! 🎉

**Pull request merged**